File: C# Fundamentals/C# OOP Basics/03. Defining Classes/Ex/01. Define a Class Person/Classes/Employee.cs
using System;
using System.Collections.Generic;
using System.Text;

public class Employee
{
    private string name;
    private int age;
    private decimal salary;
    private string position;
    private string department;
    private string email;

    public string Name
    {
        get
        {
            return this.name;
        }
        set
        {
            this.name = value;
        }
    }
    public int Age
    {
        get
        {
            return this.age;
        }
        set
        {
            this.age = value;
        }
    }
    public decimal Salary
    {
        get
        {
            return this.salary;
        }
        set
        {
            this.salary = value;
        }
    }
    public string Position
    {
        get
        {
            return this.position;
        }
        set
        {
            this.position = value;
        }
    }
    public string Department
    {
        get
        {
            return this.department;
        }
        set
        {
            this.department = value;
        }
    }
    public string Email
    {
        get
        {
            return this.email;
        }
        set
        {
            this.email = value;
        }
    }


}

File: Java/01.Advanced/01.Advanced/11.Classes/Ex/src/t_02_company_roaster/Employee.java
package t_02_company_roaster;

public class Employee implements Comparable<Employee>{
    private final String name;
    private final double salary;
    private final String position;
    private final String department;
    private final String email;
    private final int age;

    public Employee(String name, double salary, String position, String department, String email, int age) {
        this.name = name;
        this.salary = salary;
        this.position = position;
        this.department = department;
        this.email = email;
        this.age = age;
    }

    public double getSalary() {
        return salary;
    }

    public Employee(String name, double salary, String position, String department) {
        this(name, salary, position, department, "n/a", -1);
    }

    public Employee(String name, double salary, String position, String department, String email) {
        this(name, salary, position, department, email, -1);
    }

    public Employee(String name, double salary, String position, String department, int age) {
        this(name, salary, position, department, "n/a", age);
    }

    @Override
    public String toString() {
        return String.format("%s %.2f %s %d%n", this.name, this.salary, this.email, this.age);
    }

    @Override
    public int compareTo(Employee other) {
        return Double.compare(other.salary, this.salary);
    }
}
